src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
I disagree with you in that point @jason-bragg . 

The point on have multiple serializer is to declaratively select which one to use for that specific Grain/State for example...

If you have 1 serializer that will behave as proxy and inside it you will decide based on the data which one to use, you are bringing business decisions to a non-functional/infrastructure component when what you should do, is based on your business decide which infrastructure to use.

IMHO I think we should register all serializers that would be e used, and let the grain/developer when and which one to use.